webapp/channels/src/utils/constants.tsx

```diff
@@ -1902,6 +1902,7 @@ export const Constants = {
         vbscript: {name: 'VBScript', extensions: ['vbs'], aliases: ['vbs']},
         verilog: {name: 'Verilog', extensions: ['v', 'veo', 'sv', 'svh']},
         vhdl: {name: 'VHDL', extensions: ['vhd', 'vhdl'], aliases: ['vhd']},
+        vtt: {name: 'WebVTT', extensions: ['vtt'], aliases: ['vtt', 'webvtt']},
```

Comment:
Since this isn't highlighted at all, do we actually need to register this as a language? Since we have previews for text files already, I'd think this is only needed if we want to make it so that people can use WebVTT code blocks
~~~
```vtt
WEBVTT

00:01.000 --> 00:04.000
- Never drink liquid nitrogen.
```
~~~

Unfortunately, while GitHub supports this, `highlight.js` doesn't seem to, so it doesn't look as nice in MM as it does here

```vtt
WEBVTT

00:01.000 --> 00:04.000
- Never drink liquid nitrogen.
```

Reply:
@hmhealey My understanding of the way the existing logic works is that if a file type is not "highlightable" then we don't show a preview:

https://github.com/mattermost/mattermost/blob/c516d5cb153ad027aad50844e9ef7db060fd571e/webapp/channels/src/components/file_preview_modal/file_preview_modal.tsx#L346

https://github.com/mattermost/mattermost/blob/4dbf1f8c26f3a44f04ad229caa6f38aadd769dcc/webapp/channels/src/components/code_preview.tsx#L111-L113

Reply:
Ah, I didn't realize that was a requirement. I thought text files were handled separately from highlighted languages. Sorry about that!